util/src/main/java/io/kubernetes/client/util/ClientBuilder.java

```diff
@@ -209,11 +208,8 @@ public static ClientBuilder cluster() throws IOException {
     final String port = System.getenv(ENV_SERVICE_PORT);
     builder.setBasePath(host, port);
 
-    final String token =
-        new String(
-            Files.readAllBytes(Paths.get(SERVICEACCOUNT_TOKEN_PATH)), Charset.defaultCharset());
     builder.setCertificateAuthority(Files.readAllBytes(Paths.get(SERVICEACCOUNT_CA_PATH)));
-    builder.setAuthentication(new AccessTokenAuthentication(token));
+    builder.setAuthentication(new TokenFileAuthentication(SERVICEACCOUNT_TOKEN_PATH));
```

Comment:
we need to keep backward compatibility and users are expected to opt-in to the change. wdyt add a constructor w/ parameter `reloadInterval` and factor the original `cluster()` to adopt an infinite (or zero) interval?

Reply:
`cluster()` with reloading should be the default so that users dont have to do anything to migrate and add a `oldCluster` as the fail safe. 

update the InClusterClusterExample to something like:

```
// new style
ApiClient client = ClientBuilder.cluster().build();

// old style
ApiClient client = ClientBuilder.oldCluster().build();
```

WDYT?

Reply:
> cluster() with reloading should be the default so that users dont have to do anything to migrate and add a oldCluster as the fail safe.

yea we should also reflect how to opt-in SA reloading in the example. note that there's two aspects of change the pull is involving:

1. introducing capability for periodical SA reloading
2. moving default behavior the in-cluster client to 1-min reloading (preserving the old-style method or not)

the first is good while i'm leaning on doing the second in another release. i will defer to @brendandburns for thoughts

Reply:
Let's just do the whole thing. We haven't cut a `9.0.0` release yet so I think it's ok to make this change in the `9.0.0` release.